src/Uno.UI.Tests/Windows_UI_XAML_Controls/ListViewBaseTests/Given_ListViewBase_CustomContainer.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+
+namespace Uno.UI.Tests.ItemsControlTests_CustomContainer
+{
+	[TestClass]
+	public class Given_ListViewBase_CustomContainer
+	{
+		[TestMethod]
+		public void When_TemplateRootIsOwnContainer()
+		{
+			var count = 0;
+			var panel = new StackPanel();
+
+			var SUT = new MyItemsControl()
+			{
+				ItemsPanelRoot = panel,
+				ItemTemplate = new DataTemplate(() =>
+				{
+					count++;
+					return new MyCustomItemContainer() { MyValue = 42 };
+				}),
+				Template = new ControlTemplate(() => new ItemsPresenter()),
+			};
+
+			SUT.ApplyTemplate();
+
+			SUT.ItemsSource = new object[]
+			{
+				42
+			};
+
+			var container = SUT.ContainerFromIndex(0) as MyCustomItemContainer;
+
+			Assert.IsNotNull(container);
+			Assert.IsTrue(container.IsGeneratedContainer);
+			Assert.IsFalse(container.ContentTemplateRoot is MyCustomItemContainer);
+			Assert.AreEqual(42, container.DataContext);
+		}
+
+		[TestMethod]
+		public void When_TemplateSelector_RootIsOwnContainer()
+		{
+			var count = 0;
+			var panel = new StackPanel();
+
+			var itemsPresenter = new MyItemsControl();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'itemsPresenter'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5961910)